src/ax_interface/socket_io.py

```diff
@@ -25,6 +26,20 @@ def __init__(self, mib_table, run_event, loop):
 
         self.transport = self.ax_socket = None
 
+        self.ax_socket_path = constants.AGENTX_SOCKET_PATH
+        # open the snmpd config file and search for a agentx socke redefinition
+        pattern = re.compile("^agentxsocket\s+(.*)$")
+        try :
+            with open(constants.SNMPD_CONFIG_FILE,'rt') as config_file:
+                for line in config_file:
+                    for match in pattern.finditer(line):
```

Comment:
>for match in pattern.finditer(line): [](start = 20, length = 36)

why we need a loop if we are sure there is only none of one match?

Reply:
more genric IMHO, but corrected to one